include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h

```diff
@@ -118,7 +118,7 @@ exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
     internal::__buffer<policy_type, FlagType> _flags(policy, n);
     {
         auto flag_buf = _flags.get_buffer();
-        auto flags = flag_buf.template get_access<sycl::access::mode::read_write>();
+        auto flags = flag_buf.get_host_access();
```

Comment:
Essentially, we have to support the current SYCL version  and some previous versions.
To rich that goal we have the "wrappers" placed here:
https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

So, I think you should introduce a new "wrapper"  like `__dpl_sycl::get_host_access` (or something like that) for the all similar changes in that PR.

Reply:
Perhaps we can try using this one without a wrapper. In the intel/llvm/sycl repo, it's available since 2020-09 release *https://github.com/intel/llvm/blob/1fc0e4f84fb09d26ce01052127f71bb25647b507/sycl/include/CL/sycl/buffer.hpp#L293-L300)

Reply:
@MikeDvorskiy @akukanov Thanks for taking a look. Please let me know if with/w-o wrapper is the preferred way.

Reply:
Let's go ahead without the wrapper for now. If required, we can add it later.